__tests__/app.test.js

```diff
@@ -345,3 +345,57 @@ describe("POST /api/articles/:article_id/comments", () => {
   });
 });
 
+describe("GET /api/articles (queries)", () => {
+  test("status:200, returns all articles sorted by a passed in parameter", () => {
+    return request(app)
+      .get("/api/articles?sort_by=title")
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles).toBeSortedBy("title", {
+          descending: true,
+        });
+      });
+  });
+  test("status:200, returns all articles sorted by date in a specified order", () => {
+    return request(app)
+      .get("/api/articles?order=ASC")
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles).toBeSortedBy("created_at", {
+          ascending: true,
+        });
+      });
+  });
+  test("status:200, returns articles, filtered by a specified topic", () => {
+    return request(app)
+      .get("/api/articles?topic=mitch")
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles.length).toBe(11);
+      });
+  });
+  test("status:400, responds with and error when passed an invalid sort by query", () => {
+    return request(app)
+      .get("/api/articles?sort_by=titlegg")
+      .expect(400)
+      .then((res) => {
+        expect(res.body.msg).toBe("Bad request");
+      });
+  });
+  test("status:400, responds with and error when passed an invalid order by query", () => {
+    return request(app)
+      .get("/api/articles?order=balls")
+      .expect(400)
+      .then((res) => {
+        expect(res.body.msg).toBe("Bad request");
+      });
+  });
+  test("status:404, responds with an error message when article does not exist", () => {
+    return request(app)
+      .get("/api/articles?topic=nothing")
+      .expect(404)
+      .then((res) => {
+        expect(res.body.msg).toBe("Resource not found");
+      });
+  });
+});
```

Comment:
What if the path was `/api/articles?notvalidsort=nope`? 

Also add a test that makes the three queries all at the same time.